### PR TITLE
Address some fides lang 1.4 test failures

### PR DESCRIPTION
**Pull request by pattisdr**
Closes #<issue>

### Code Changes

* [ ] _list your code changes here_

### Steps to Confirm

* [ ] _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

Addresses most of the failing tests in https://github.com/ethyca/fides/pull/3343

### CODE REVIEW

File: src/fides/api/ctl/migrations/versions/63b482f5b49b_upgrade_to_fideslang_1_4.py [hunk 1]
```diff
@@ -22,9 +22,9 @@
 
 # The `key` is the old value, the `value` is the new value
 data_use_upgrades: Dict[str, str] = {
-    "third_party_sharing.payment_processing": "essential.service.payment_processing"
-    "third_party_sharing.fraud_detection",
-    "essential.service.fraud_detection"
+    "third_party_sharing.payment_processing": "essential.service.payment_processing",
+    "third_party_sharing.fraud_detection": "essential.service.fraud_detection",
+    "advertising": "marketing.advertising",
```
Comment (pattisdr) [inline src/fides/api/ctl/migrations/versions/63b482f5b49b_upgrade_to_fideslang_1_4.py:27]: Making some assumptions here -

File: tests/ops/api/v1/endpoints/test_consent_request_endpoints.py [hunk 2]
```diff
@@ -914,7 +914,7 @@ def test_set_consent_consent_preferences(
         assert response.status_code == 200
         expected_consent_data: list[dict[str, Any]] = [
             {
-                "data_use": "advertising",
+                "data_use": "marketing.advertising",
                 "data_use_description": None,
                 "opt_in": True,
```
Comment (pattisdr) [inline tests/ops/api/v1/endpoints/test_consent_request_endpoints.py:919]: I think this test is failing for ordering reasons

File: tests/ctl/core/test_evaluate.py [hunk 3]
```diff
@@ -580,12 +580,12 @@ def test_failed_evaluation_error_message(
...
                               '(aggregated.anonymized.unlinked_pseudonymized.pseudonymized.identified) '
                               'for data uses '
-                              '(third_party_sharing.payment_processing) and '
+                              '(third_party_sharing) and '
                               'subjects (customer)',
                     'violating_attributes': { 'data_categories': [ 'user.political_opinion'],
                                               'data_qualifier': 'aggregated.anonymized.unlinked_pseudonymized.pseudonymized.identified',
                                               'data_subjects': ['customer'],
-                                              'data_uses': [ 'third_party_sharing.payment_processing']}}]}
+                                              'data_uses': [ 'third_party_sharing']}}]}
```
Comment (pattisdr) [inline tests/ctl/core/test_evaluate.py:588]: Haven't gotten this test quite right -

Comment (ThomasLaPiana) [reply tests/ctl/core/test_evaluate.py:588]: no worries, I'll take a look!

File: tests/ops/service/connectors/test_consent_email_connector.py [hunk 4]
```diff
@@ -547,7 +553,7 @@ def test_test_connection_call(
 
         assert (
             preferences[0]["consent_preferences"][1]["data_use"]
-            == "Improve the capability"
+            == "Improves the product, service, application or system."
```
[the remark this answers is not in the document]
Comment (ThomasLaPiana) [reply tests/ops/service/connectors/test_consent_email_connector.py:556]: seems like a question for @mfbrown 🤔

**Comment by ThomasLaPiana** [comment]
@pattisdr I'm going to merge this into my 1.4 branch so we can view all of the changes required in one place 🙂 

Thank you so much for working on these!

**Pull request merged**